local-game.js

```diff
@@ -0,0 +1,512 @@
+			erase_roads(selectedPiece);
+	}
+	selectedPiece = piece;
+	var i,j; 
+	for ( j = 1; j <= 12; j++){
+		if(the_checker[j].id == piece){
+			i = j;
+			selectedPieceindex = j;
+			match = true;
+		}
+	}
+
+	if(oneMove && !attackMoves(oneMove)){
+		changeTurns(oneMove);
+		oneMove = undefined;
+		return false;
+	}
+	if(oneMove && oneMove != the_checker[i] ){
+		return false;
+	}
+
+	if(!match) {
+	 return 0 ; 
+	}
+
+	
+	if(the_checker[i].color =="white"){
+		tableLimit = 8;
+		tableLimitRight = 1;
+		tableLimitLeft = 8;
+		moveUpRight = 7;
+		moveUpLeft = 9;
+		moveDownRight = - 9;
+		moveDownLeft = -7;
+	}
+	else{
+		tableLimit = 1;
+		tableLimitRight = 8;
+		tableLimitLeft = 1;
+		moveUpRight = -7;
+		moveUpLeft = -9;
+		moveDownRight = 9;
+		moveDownLeft = 7;
+	}
+ 	
+
+
+		attackMoves(the_checker[i]); 
+	
+
+	
+
+ 	if(!mustAttack){
+ 	  downLeft = checkMove( the_checker[i] , tableLimit , tableLimitRight , moveUpRight , downLeft);
+		downRight = checkMove( the_checker[i] , tableLimit , tableLimitLeft , moveUpLeft , downRight);
+		if(the_checker[i].king){
+			upLeft = checkMove( the_checker[i] , reverse_tableLimit , tableLimitRight , moveDownRight , upLeft);
+			upRight = checkMove( the_checker[i], reverse_tableLimit , tableLimitLeft , moveDownLeft, upRight)
+		}
+	}
+	if(downLeft || downRight || upLeft || upRight){
+			return true;
+		}
+	return false;
+	
+}
+
+
+function erase_roads(piece){
+	if(downRight) block[downRight].id.style.background = "#BA7A3A";
+	if(downLeft) block[downLeft].id.style.background = "#BA7A3A";
+	if(upRight) block[upRight].id.style.background = "#BA7A3A";
+	if(upLeft) block[upLeft].id.style.background = "#BA7A3A";
+}
+		
+
+
+function makeMove (index) {
+	var isMove = false;
+	if(!selectedPiece) 
+		return false;
+	if(index != upLeft && index != upRight && index != downLeft && index != downRight){
+		erase_roads(0);
+		selectedPiece = undefined;
+		return false;
+	}
+
+	if(the_checker[1].color=="white"){
+		cpy_downRight = upRight;
+		cpy_downLeft = upLeft;
+		cpy_upLeft = downLeft;
+		cpy_upRight = downRight;
+	}
+	else{
+		cpy_downRight = upLeft;
+		cpy_downLeft = upRight;
+		cpy_upLeft = downRight;
+		cpy_upRight = downLeft;
+	}  
+
+	if(mustAttack)  
+		multiplier = 2;
+	else
+		multiplier = 1;
+
+
+		if(index == cpy_upRight){
+			isMove = true;		
+			if(the_checker[1].color=="white"){
+				
+				executeMove( multiplier * 1, multiplier * 1, multiplier * 9 );
+				
+				if(mustAttack) eliminateCheck(index - 9);
+			}
+			else{
+				executeMove( multiplier * 1, multiplier * -1, multiplier * -7);
+				if(mustAttack) eliminateCheck( index + 7 );
+			}
+		}
+
+		if(index == cpy_upLeft){
+	
+			isMove = true;
+			if(the_checker[1].color=="white"){
+				executeMove( multiplier * -1, multiplier * 1, multiplier * 7);
+			 	if(mustAttack)	eliminateCheck(index - 7 );				
+			}
+			else{
+				executeMove( multiplier * -1, multiplier * -1, multiplier * -9);
+				if (mustAttack) eliminateCheck( index + 9 );
+			}
+		}
+

```

Comment:
My suggestion is to refactor this and to describe all possible steps for king at first and than add limitations for  if !king. The idea is to get more readable code